pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,383 @@
+	strimziAddonIdQE        = "managed-kafka-qe"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "register-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.registerCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.registerCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.registerCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.registercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterId.description"))
+	flags.IntVar(&opts.pageNumber, "page-number", int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber)), f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageNumber.description"))
+	flags.IntVar(&opts.pageSize, "page-size", 100, f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageSize.description"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) (err error) {
+	opts.pageNumber = int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber))
+	err = getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	if len(opts.clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.registerCluster.run.noClusterFound")
+	}
+
+	if opts.selectedClusterId == "" {
+		err = runClusterSelectionInteractivePrompt(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range opts.clusterList {
+			cluster := opts.clusterList[i]
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
+	}
+	err = getOrCreateMachinePoolList(opts)
+	if err != nil {
+		return err
+	}
+	err = selectAccessPrivateNetworkInteractivePrompt(opts)
+	if err != nil {
+		return err
+	}
+	err = registerClusterWithKasFleetManager(opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func getPaginatedClusterList(opts *options) error {
+	cl, err := clustermgmt.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, opts.pageNumber, opts.pageSize)
+	if err != nil {
+		return err
+	}
+
+	opts.clusterList = validateClusters(cl, opts.clusterList)
+	return nil
+}
+
+func validateClusters(clusters *clustersmgmtv1.ClusterList, cls []clustersmgmtv1.Cluster) []clustersmgmtv1.Cluster {
+	for _, cluster := range clusters.Slice() {
+		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
+			cls = append(cls, *cluster)
+		}
```

Comment:
This can be done when fetching the list of cluster. 
the `/api/clusters_mgmt/v1/clusters` takes a search parameter and we could search for `multi_az='t' AND state='ready'` this could also help reduce the page size in case the user is in a org with a lot of single az clusters or clusters that are not ready